### PR TITLE
Use new metadata API

### PR DESCRIPTION
**Pull request by cboulanger**

### CODE REVIEW

**Review by hkollmann — DISMISSED** [review]

**Comment by cboulanger** [comment]
@johnspackman Can you review and merge if you approve of the PR?

**Comment by hkollmann** [comment]
I think we can remove getQoodooVersion instead of deprecate it, could‘nt we?

File: lib/qxcli/commands/Command.js [hunk 1]
```diff
@@ -128,37 +129,43 @@ qx.Class.define("qxcli.commands.Command", {
...
+      if( ! semver.valid( qxversion) ){
+        throw new Error(`Invalid qooxdoo version ${qxversion}. Must be valid semver string.`);
+      }
+      return qxversion;
     },
     
     /**
      * Given the path to a qooxdoo framework folder, returns the qooxdoo version 
      * in semver-compatible format.
+     * @deprecated Use {@link #getUserQxVersion} instead.
```
Comment (johnspackman) [inline lib/qxcli/commands/Command.js]: I agree with @hkollmann there's no need to deprecate at this stage, we're still officially in beta and the API is allowed to change without warning.

**Review by johnspackman — CHANGES_REQUESTED** [review]

**Comment by hkollmann** [comment]
In this case you must change template_vars.js, line 69 too!

**Comment by cboulanger** [comment]
I tried to remove the method and there was a problem which I did not have the time to investigate - that's why I left it and deprecated it for the time being. I agree with you and will have a second look hopefully later today.

**Comment by cboulanger** [comment]
@hkollmann @johnspackman Ok, I have reworked it so that now the version number is no longer taken from the `versions.txt` file but from the Manifest.json (as it should be). For this to work, you need the newest master with  @hkollmann's PR that bumped the version numbers (otherwise it the version is still `5.1`, which is invalid semver in any case).

**Review by hkollmann — APPROVED** [review]

**Comment by cboulanger** [comment]
@johnspackman can you review so that this can get merged? Thanks!

**Review by johnspackman — APPROVED** [review]

**Pull request merged**